branchpro/models.py

```diff
@@ -42,3 +43,119 @@ def simulate(self, parameters, times):
             value appears in ``times``.
         """
         raise NotImplementedError
+
+#
+# BranchProModel Class
+#
+
+
+class BranchProModel(ForwardModel):
+    r"""BranchProModel Class:
+    Class for the models following a Branching Processes behaviour.
+    It inherits from the `ForwardModel`` class.
+
+    In the branching process model, we track the number of cases
+    registered each day, I_t, also known as the "incidence" at time t.
+
+    The incidence at time t is a random variable distributed according to
```

Comment:
Just to stress that this is a model and the incidence doesn't truly follow a Poisson distribution, could you replace "is" by "model by", I.e. 
"The incidence number at time t is modelled by a random ..."